### PR TITLE
Add pre-commit hooks

### PR DESCRIPTION
**Pull request by andersy005**
- Towards #8

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+[tool.black]
+line-length = 100
+target-version = ['py38']
+skip-string-normalization = true
```
Comment (dcherian) [inline pyproject.toml:4]:
```suggestion
```

this will reduce this diff by a lot

File: setup.cfg [hunk 2]
```diff
@@ -5,3 +5,24 @@ versionfile_source = cupy_xarray/_version.py
...
+[flake8]
+exclude = docs,versioneer.py,cupy_xarray/_version.py
+ignore = E203,E266,E501,W503,E722,E402,C901,E731
+max-line-length = 100
+max-complexity = 18
+select = B,C,E,F,W,T4,B9
+
+[isort]
+known_first_party=cupy_xarray
+known_third_party=
```
Comment (dcherian) [inline setup.cfg:18]: a bunch of this can be replaced by `profile=black` I think

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -0,0 +1,50 @@
+ci:
+  autoupdate_schedule: monthly
```
Comment (dcherian) [inline .pre-commit-config.yaml:2]:
```suggestion
  autoupdate_schedule: quarterly
```

I prefer this but not very strongly. It's nice to not get pinged for updates that make no difference.

File: cupy_xarray/accessors.py [hunk 4]
```diff
@@ -99,7 +93,7 @@ def get(self):
         return self.da.data.get()
 
 
-@register_dataset_accessor("cupy")
+@register_dataset_accessor('cupy')
```
Comment (dcherian) [inline cupy_xarray/accessors.py:96]: hmmm why didnt this change back

**Review by dcherian — APPROVED** [review]

File: cupy_xarray/accessors.py (hunk 4 above)
Comment (andersy005) [reply cupy_xarray/accessors.py]: should be fixed now.

**Pull request merged**